storage/shard.go

```diff
@@ -786,15 +854,29 @@ func (s *dbShard) Bootstrap(
 
 	multiErr := xerrors.NewMultiError()
 	for _, dbBlocks := range bootstrappedSeries {
-		entry, err := s.writableSeries(dbBlocks.ID)
+		// First lookup if series already exists
+		entry, _, err := s.tryRetrieveWritableSeries(dbBlocks.ID)
 		if err != nil {
 			multiErr = multiErr.Add(err)
 			continue
 		}
+		if entry == nil {
+			// Synchronously insert to avoid waiting for
+			// the insert queue potential delayed insert
+			entry, err = s.insertSeriesSync(dbBlocks.ID, insertSyncIncWriterCount)
+			if err != nil {
+				multiErr = multiErr.Add(err)
+				continue
+			}
+		}
 
 		err = entry.series.Bootstrap(dbBlocks.Blocks)
+		if err != nil {
```

Comment:
Sure thing, that's how it was before - just wanted to be more Go-ish but I'll change this back =]